integration_tests/def_func_01.py

```diff
@@ -0,0 +1,76 @@
+from lpython import i32,i64
+
+def factorial_1(x: i32, y:i32 =1) ->i32 :
+    if x <= 1:
+         return y
+    return x * factorial_1(x-1)
+
+def factorial_2(x: i32, y:i32=3 ,z:i32 =2) ->i32:
+    if x ==4:
+        return x * y * z
+    return x * factorial_2(x-1)
+
+def default_func(x : str ="Hello" ,y : str = " ", z : str = "World") ->str:
+    return x + y + z
+
+
+def even_positons(iterator : i32 , to_add : str = "?")-> str:
+    if (iterator == 10): return ""
+
+    if iterator%2 == 0 :
+        return to_add + even_positons(iterator+1,"X")
+
+    return to_add +even_positons(iterator+1)
+
+
+
+def test_all():
+
+    test_01 : i32  = factorial_1(5,0)
+    print("test_01 is =>",test_01)
+    assert test_01 == 120
+
+    test_02 : i32  = factorial_1(1,5555)
+    print("test_02 is =>",test_02)
+    assert test_02 == 5555
```

Comment:
Add a test for not passing in the second argument and ensure it returns the default value "1".